client/components/MovieGrid.js

```diff
@@ -11,23 +11,74 @@ Container Component For Grid data
 Uses Grid as presentation component
 */
 
-export const MovieGrid = () => {
-    const [movies, setMovies] = useState([]);
+// const Child = forwardRef((props, ref) => {
 
-    useEffect(()=> {
-        // console.log(getTopMovies());
+//     // The component instance will be extended
+//     // with whatever you return from the callback passed
+//     // as the second argument
+//     useImperativeHandle(ref, () => ({
+  
+//       getAlert() {
+//         alert("getAlert from Child");
+//       }
+  
+//     }));
+
+export const MovieGrid = forwardRef((props, ref) => {
+        const [movies, setMovies] = useState([]);
+
+    function getTopMoviesApi(){
         let movieData = [];
         getTopMovies().then(data => {
             data.results.forEach((x, i) => movieData.push({"Movie": x.title, "Overview": x.overview, "Release Date": x.release_date, "Language": x.original_language}));
             setMovies(movieData);
         })
         .catch(err => console.log(err));
+    }
+
+    function searchMoviesApi(searchQuery){
+        let movieData = [];
+        searchMovies(searchQuery).then(data => {
+            data.results.forEach((x, i) => movieData.push({"Movie": x.title, "Overview": x.overview, "Release Date": x.release_date, "Language": x.original_language}));
+            setMovies(movieData);
+        }).catch(err => console.log(err));
+        console.log(movieData);
+    }
+
+
+
+    useEffect(()=> {
+        // console.log(getTopMovies());
+        getTopMoviesApi();
         
-    },[])
+    },[]);
+
+    useImperativeHandle(ref, () => ({
```

Comment:
Interesting! Can you walk me through where the original ref is passed in from, and how this useImperativeHandle gets called? Basically I'm looking for an onPress somewhere, and tryina check out what's going on 🤔